LoopKit/GlucoseKit/GlucoseStore.swift

```diff
@@ -143,511 +150,496 @@ public final class GlucoseStore: HealthKitSampleStore {
-            case .success(let cacheDeletedCount):
-                if cacheDeletedCount > 0 {
-                    cacheChanged = true
-                }
-                self.log.debug("Finished deletion: HK delete count = %d, cache delete count = %d", deleted.count, cacheDeletedCount)
-            case .failure(let error):
-                self.log.error("Samples deleted from HK could not be deleted from cache: %{public}@", String(describing: error))
-                completion(false)
+            if !changed {
+                completion(true)
                 return
             }
 
-            if let startDate = newestSampleStartDateAddedByExternalSource {
-                self.purgeOldGlucoseSamples(includingManagedDataBefore: startDate)
-            }
-
-            if cacheChanged {
-                self.updateLatestGlucose()
-            }
-
-            if samplesAddedByExternalSourceWithinManagedDataInterval {
-                NotificationCenter.default.post(name: GlucoseStore.glucoseSamplesDidChange, object: self, userInfo: [GlucoseStore.notificationUpdateSourceKey: UpdateSource.queriedByHealthKit.rawValue])
+            // Purge expired managed data from HealthKit
+            if let newestStartDate = added.map({ $0.startDate }).max() {
+                self.purgeExpiredManagedDataFromHealthKit(before: newestStartDate)
             }
 
+            self.notifyUpdatedGlucoseData(updateSource: .queriedByHealthKit)
             completion(true)
         }
     }
 }
 
+// MARK: - Fetching
+
 extension GlucoseStore {
-    /// Add new glucose values to HealthKit.
-    ///
-    /// This operation is performed asynchronously and the completion will be executed on an arbitrary background queue.
+    /// Retrieves glucose samples within the specified date range.
     ///
     /// - Parameters:
-    ///   - glucose: A glucose sample to save
-    ///   - completion: A closure called after the save completes
-    ///   - result: The saved glucose value
-    public func addGlucose(_ glucose: NewGlucoseSample, completion: @escaping (_ result: GlucoseStoreResult<GlucoseValue>) -> Void) {
-        addGlucose([glucose]) { (result) in
-            switch result {
-            case .success(let values):
-                completion(.success(values.first!))
-            case .failure(let error):
-                completion(.failure(error))
-            }
+    ///   - start: The earliest date of glucose samples to retrieve, if provided.
+    ///   - end: The latest date of glucose samples to retrieve, if provided.
+    ///   - completion: A closure called once the glucose samples have been retrieved.
+    ///   - result: An array of glucose samples, in chronological order by startDate, or error.
+    public func getGlucoseSamples(start: Date? = nil, end: Date? = nil, completion: @escaping (_ result: Result<[StoredGlucoseSample], Error>) -> Void) {
+        queue.async {
```

Comment:
[nit] This is a small nit observation, and I may not have all the pieces understood here, but it looks like the private `getGlucoseSamples` below blocks in `performAndWait`, and then results are passed through `completion` here.  Seems fine and should work, but won't it potentially block `queue`?  Why not just pass the `completion` to it, and have the `getGlucoseSamples` function below do a regular `perform` and not hop queues at all?  

Again, I may not have the whole picture here, and it is a minor comment.